libr/core/cio.c

```diff
@@ -106,6 +106,38 @@ R_API bool r_core_dump(RCore *core, const char *file, ut64 addr, ut64 size, int
 	return true;
 }
 
+static int __endian_swap(ut8 *buf, ut32 blocksize, ut8 len) {
+	ut32 i;
+	ut8 tmp;
+	ut16 v16;
+	ut32 v32;
+	ut64 v64;
+	if (len != 8 && len != 4 && len != 2 && len != 1) {
+		eprintf ("Invalid word size. Use 1, 2, 4 or 8\n");
+		return false;
+	}
+	if (len == 1) {
+		return true;
+	}
+	for (i = 0; i < blocksize; i += len) {
+		switch (len) {
+		case 8:
+			v64 = r_read_be64(buf+i);
```

Comment:
Also theres a r read function that takes a size arg and a endian arg that does exactly this iirc

Reply:
This code can do oobreada because the lengh of the buffer doesnt needs to be aligned to the word size

Reply:
I will double check that. It was not a problem yesterday.